### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by smigueles**
# Summary

- API Services integration
- An user can get, post, put and delete by an http method

# Test of Case
N/A

# Screenshots
Captura de pantalla de 2021-06-14 12-02-05

Captura de pantalla de 2021-06-14 12-32-43

# Trello Issue

[https://trello.com/c/dOuJ2ev7/25-agregando-servicios-a-la-app](url)

### CODE REVIEW

File: src/components/CardExpression/index.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-import {View, Text, TextInput, TouchableOpacity} from 'react-native';
+import {View, Text, TextInput, TouchableOpacity, Alert} from 'react-native';
 import {useDispatch} from 'react-redux';
 
 import {styles} from './style';
-import actionsCreator from '../../redux/actions';
+import historyAction from '../../redux/historyApi/actions';
 import {RULES} from '../../constants/rules';
 
 const CardExpression = ({n}) => {
```
Comment (taltrui) [inline src/components/CardExpression/index.js]: What is n??? Be more expressive with names.

File: src/redux/historyApi/reducer.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    case actions.DELETE_EXPRESSION_BY_ID: {
+      return {...state, deleteExpressionLoading: true};
+    }
+    case actions.DELETE_EXPRESSION_BY_ID_SUCCESS: {
+      return {
+        ...state,
+        deleteExpressionLoading: false,
+        deleteExpressionError: null,
+        deleteExpressionMsg: action.payload,
+        idMsg: Math.random() * 100,
```
Comment (taltrui) [inline src/redux/historyApi/reducer.js]: What is this msg for? Also, using a non-deterministic function inside a reducer, which should be deterministic and pure is a big no-no.

File: src/redux/historyApi/reducer.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    case actions.DELETE_ALL_EXPRESSIONS: {
+      return {...state, deleteAllExpressionLoading: true};
+    }
+    case actions.DELETE_ALL_EXPRESSIONS_SUCCESS: {
+      return {
+        ...state,
+        deleteAllExpressionLoading: false,
+        deleteAllExpressionError: null,
+        deleteAllExpressionMsg: action.payload,
+        idMsg: Math.random() * 100,
```
Comment (taltrui) [inline src/redux/historyApi/reducer.js]: Same here

File: src/redux/historyApi/reducer.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    case actions.EDIT_EXPRESSION: {
+      return {...state, editExpLoading: true};
+    }
+    case actions.EDIT_EXPRESSION_SUCCESS: {
+      return {
+        ...state,
+        editExpLoading: false,
+        editExpError: null,
+        editExpMsg: action.payload,
+        idMsg: Math.random() * 100,
```
Comment (taltrui) [inline src/redux/historyApi/reducer.js]: And here

File: src/screens/Display/index.js [hunk 5]
```diff
@@ -1,18 +1,36 @@
...
+      headerRight: () => (
+        <TouchableOpacity
+          style={styles.btnLogOut}
+          onPress={() => dispatch(authAction.logOut(navigation))}>
+          <Text style={styles.logOutTxt}>Logout</Text>
+        </TouchableOpacity>
+      ),
+    });
+  });
+
```
Comment (taltrui) [inline src/screens/Display/index.js]: Why this set options? why not directly have the logout in the config of the screen when declaring it?

File: src/screens/Record/index.js [hunk 6]
```diff
@@ -1,23 +1,28 @@
...
+  return {history, user, historyApi};
 };
 
-const Record = ({history, navigation}) => {
+const Record = ({historyApi, navigation}) => {
   const dispatch = useDispatch();
 
+  useEffect(() => {
+    dispatch(historyAction.getExpressions());
+  }, [historyApi.idMsg]);
```
Comment (taltrui) [inline src/screens/Record/index.js]: This is wrong here, you should get the expressions in the edit/update action, after the success action. Doing this is not a really good approach.

**Review by taltrui — CHANGES_REQUESTED** [review]

**Comment by smigueles** [comment]
Changes applied, waiting for a new revision

**Review by lucawidergy — APPROVED** [review]
esta ok, algunos comentario para tener en cuenta para la integracion con redux

1- cuando hago logout, no deberia quedarme cargado el user y la password. 
2- me registre, hice log out y apareci en la pantalla de regitracion y no en login.
3- Validar que haya password/email antes de hacer submit!
4- un loading para el login/logpout
5- en registracion, pone que tengan q confirmar la contraseña y checkea que sean iguales.
6- indicar en los input que es cada cosa jajaa

**Review by taltrui — APPROVED** [review]

**Pull request merged**